src/transformers/tokenization_utils.py

```diff
@@ -199,10 +213,16 @@ def _add_tokens(self, new_tokens: Union[List[str], List[AddedToken]], special_to
 
         # Make sure we don't split on any special tokens (even they were already in the vocab before e.g. for Albert)
         if special_tokens:
-            self.unique_no_split_tokens = sorted(set(self.unique_no_split_tokens).union(set(new_tokens)))
+            if len(new_tokens) == 1:
+                _insert_one_token_to_ordered_list(self.unique_no_split_tokens, new_tokens[0])
```

Comment:
I think we're losing the `sorted` on the `self.unique_no_split_token` here, which is required by the `_insert_one_token_to_ordered_list`, right?

Reply:
Hi @LysandreJik , `_insert_one_token_to_ordered_list ` uses `bisect` to find a position in the sorted list to insert a new token. So as long as the input list is already sorted, it remains sorted after the insertion. To make sure, I added the following assertion to the benchmark script to test:

```python
# after loading tokenizer from a saved model
all_values = tokenizer.unique_no_split_tokens
assert all(all_values[i+1] > all_values[i] for i in range(len(all_values) - 1))
```

Reply:
The assumption that the input list is already sorted is because it would have been built using this which sorts values? Other than that particular question it looks good to me. I'll ping another from the team and if it's good for them we'll merge. Thanks for your patience!

Reply:
Yeah, that's right. Because populating an empty `tokenizer. unique_no_split_tokens` with this `bisect` approach ensures it will always be sorted. Thanks!